### DISCUSSION TITLE
Hydration for create_server_action?

### DISCUSSION THREAD

**Discussion by y-haidar** [leptos-rs/leptos#1490] [Q&A] [answered]
Hi I have been playing around with leptos and loved the `Progressive Enhancement (and Graceful Degradation)` concept, and so I have been trying to use `ActionForm`s for all the inputs, but the downside that I noticed is that if a page is loaded, the action's `value()` will be empty until the `wasm` is loaded and then manually dispatch the action.

Is there a way to enable hydration/ssr for it? Am I missing something?

Bellow is sample code for what I am trying to do:

```rust
#[component]
pub fn MonthlyView(
  cx: Scope,
  data: RwSignal<Option<Result<Vec<(NaiveDate, Vec<String>)>, ServerFnError>>>,
) -> impl IntoView {
  // ...
}

// This included in the Router
#[component]
pub fn TestPage(cx: Scope) -> impl IntoView {
  let today = chrono::Utc::now().naive_utc().date();
  let test_action = create_server_action::<GetMonthlyView>(cx);

  // MonthlyView IS EMPTY until this runs(or user submits a request)
  test_action.dispatch(GetMonthlyView {
    year: today.year() as u32,
    month: today.month(),
  });

  view! { cx,
    <YearMonthInput action=test_action/>
    <MonthlyView data=test_action.value()/>
  }
}
```

The `YearMonthInput` component basically contains the `ActionForm`.

**Comment by gbj** [comment, marked as answer]
If you ever find yourself creating an action and immediately dispatching it, you should probably be using a resource instead. If you used a resource to load the data here, it would load on the server and stream down to the client. Any data you used from it in your view behind a Suspense or Transition would appear in the HTML before the WASM had loaded.

You could then use a `create_effect` and `Resource::set` to feed the action's value into the resource.

**Reply by y-haidar** [reply]
Worked perfectly, didn't know about the `Resource::set`. I was using a resource initially, but only figured how to change it via signals.

Also love your work thank you for leptos :)

**Reply by y-haidar** [reply]
Also a regular `<Form method="GET ...> ... </Form>` and a resource with `use_query_map` as oppose to `create_signal` was what I wanted, no `create_server_action` is needed. I am clearly not a forntend guy
